crates/common/tedge_config/src/tedge_toml/tedge_config_location.rs

```diff
@@ -346,6 +344,14 @@ fn update_with_environment_variables(
     Ok(())
 }
 
+fn parse_key_without_warnings(tedge_key: &str) -> Result<WritableKey, ParseKeyError> {
+    tracing::subscriber::with_default(NoSubscriber::new(), || {
+        tedge_key
+            .to_ascii_lowercase()
+            .parse::<tedge_config::WritableKey>()
+    })
```

Comment:
question: is the warning ultimately emitted by figment and that's why we can't just not emit it in the first place?

`WritableKey` parse impl is emitted by a macro, so I can't seem to find it and confirm it myself.

Reply:
https://github.com/thin-edge/thin-edge.io/blob/1ff8cf459af0ba4a463c460774b9e968abd1f5ef/crates/common/tedge_config_macros/impl/src/query.rs#L1109-L1126 shows what this expands to in a simple case

Reply:
The deprecation warning is shown when running e.g. `tedge config get mqtt.external.certfile` (or it is as of 64dc06e, turns out there was also a bug in logging the deprecated keys when running `tedge config ...`), so it is useful.